### PR TITLE
Fix cross-references in examples.

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Not everything works due to missing upstream documentation or missing setup in our code, but I only touched the `examples` directory for now.

## PR Checklist

- [N/A] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [N/A] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: doc/missing-references.json [hunk 1]
```diff
@@ -3233,149 +3216,40 @@
...
-    "mpl_toolkits.axes_grid1.inset_axes": [
-      "doc/gallery/axes_grid1/demo_colorbar_with_inset_locator.rst:14"
-    ],
-    "scatter": [
-      "doc/gallery/axisartist/demo_floating_axes.rst:16"
-    ],
-    "GridHelperCurveLinear": [
-      "doc/gallery/axisartist/demo_floating_axes.rst:17",
-      "doc/gallery/axisartist/demo_floating_axes.rst:18",
+    "floating_axes.FloatingSubplot": [
```
Comment (QuLogic) [inline doc/missing-references.json]: New because for some reason this isn't in the documentation.

File: doc/missing-references.json [hunk 2]
```diff
@@ -3233,149 +3216,40 @@
...
-    "xy=(x, y)": [
-      "doc/gallery/text_labels_and_annotations/annotation_demo.rst:39"
-    ],
-    "xytext=(x, y)": [
-      "doc/gallery/text_labels_and_annotations/annotation_demo.rst:39"
-    ],
-    "font_manager.FontProperties": [
-      "doc/gallery/text_labels_and_annotations/font_file.rst:14"
-    ],
     "ConciseDateConverter": [
```
Comment (QuLogic) [inline doc/missing-references.json]: This doesn't appear in `matplotlib.dates` documentation.

File: examples/misc/ftface_props.py [hunk 3]
```diff
@@ -3,9 +3,9 @@
 Font properties
 ===============
 
-This example lists the attributes of an `FT2Font` object, which describe global
-font properties.  For individual character metrics, use the `Glyph` object, as
-returned by `load_char`.
+This example lists the attributes of an `.FT2Font` object, which describe
```
Comment (QuLogic) [inline examples/misc/ftface_props.py]: Not yet fixed because our C extensions don't seem to be in the documentation.

**Comment by timhoffm** [comment]
Not attempting to backport as that will likely cause conflicts and 3.2 is not too far away.

**Pull request merged**